packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx

```diff
@@ -62,14 +65,102 @@ const defaults = {
   theme: 'light',
 };
 
+interface Column {
+  Header?: string;
+  accessor?: string | (() => void);
+  Cell?: () => void; // optional cell formatter
+}
+
+type Size = 'xs' | 'sm' | 'md' | 'lg';
+
+interface DataSpreadsheetProps {
+  /**
+   * Specifies the cell height
+   */
+  cellSize?: Size;
+
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className?: string;
+
+  /**
+   * The data that will build the column headers
+   */
+  columns?: readonly Column[];
+
+  /**
+   * The spreadsheet data that will be rendered in the body of the spreadsheet component
+   */
+  data?: readonly object[];
+
+  /**
+   * Sets the number of empty rows to be created when there is no data provided
+   */
+  defaultEmptyRowCount?: number;
+
+  /**
+   * The spreadsheet id
+   */
+  id?: number | string;
+
+  /**
+   * The event handler that is called when the active cell changes
+   */
+  onActiveCellChange?: () => void;
+
+  /**
+   * The setter fn for the data prop
+   */
+  onDataUpdate?: ({ ...args }) => void;
+
+  /**
+   * The event handler that is called when the selection area values change
+   */
+  onSelectionAreaChange?: () => void;
+
+  /**
+   * The aria label applied to the Select all button
+   */
+  selectAllAriaLabel: string;
+
+  /**
+   * The aria label applied to the Data spreadsheet component
+   */
+  spreadsheetAriaLabel: string;
+
+  /**
+   * The theme the DataSpreadsheet should use (only used to render active cell/selection area colors on dark theme)
+   */
+  theme?: 'light' | 'dark';
+
+  /**
+   * The total number of columns to be initially visible, additional columns will be rendered and
+   * visible via horizontal scrollbar
+   */
+  totalVisibleColumns?: number;
+
+  /* TODO: add types and DocGen for all props. */
+}
+
+type ActiveCellCoordinates = {
+  column?: string | number;
+  row?: string | number;
+};
+
+type PrevStateType = {
+  activeCellCoordinates?: ActiveCellCoordinates;
+  isEditing?: boolean;
+};
+
 /**
  * DataSpreadsheet: used to organize and display large amounts of structured data, separated by columns and rows in a grid-like format.
  */
 export let DataSpreadsheet = React.forwardRef(
   (
     {
       // The component props, in alphabetical order (for consistency).
-      cellSize = defaults.cellSize,
+      cellSize = 'sm',
```

Comment:
i believe this should still be `defults.cellSize`

Reply:
The type of `cellSize` changed to `type Size = 'xs' | 'sm' | 'md' | 'lg'`, hence we can define the default `cellSize` as one of type `Size` not a string type.